src/Pool.sol

```diff
@@ -19,7 +19,7 @@ interface IDai is IERC20 {
         uint8 v,
         bytes32 r,
         bytes32 s
-    ) external;
+    ) virtual external;
```

Comment:
Huh, why? Isn't everything in an interface virtual by definition?

Reply:
Ah, yes. The `override` for the permit method was enough. More checks in higher Solidity versions :)